### ISSUE TITLE
GPL-853 - Backload eagle samplesinto  lighthouse_sample table (PSDTHV)

### ISSUE THREAD

**Issue by KatyTaylor** [sanger/General-Backlog-Items#113]
**Describe the bug**
Eagle samples presence in the lighthouse_sample table in MLWH is not consistent, which could cause issues with reporting.  We currently have 3 different scenarios:

1. (as per MK) Plate map data arrives by email, uploaded by manifest. Not in Mongo / lighthouse_sample table at all.
2. (as per Cam) Plate map data arrives by sftp, uploaded by manifest. In Mongo / lighthouse_sample table, but COG UK IDs not set in lighthouse_sample table
3. (as per AP) Plate map data arrives by sftp, Sentinel or Beckman process used. In Mongo / lighthouse_sample table, COG UK ID set but 'cohort' field not filled out. Won't pick samples that fail the current business 'fit to pick' rules.

We want:
- data in Mongo & lighthouse sample (as per all Lighthouse samples)
- COG UK IDs propagated to lighthouse_sample (as per all Lighthouse samples)
- cohort field filled in (Eagle samples only)
- all Eagle samples picked regardless of whether they pass fit to pick rules

**Comment by KatyTaylor** [comment]
Didn't think this through properly -
If the samples come in on manifests and the plate map data has been emailed then presumably it hasn't been processed by crawler and therefore the samples aren't in lighthouse_sample table at all? Is this a problem?

**Comment by KatyTaylor** [comment]
Update:
We are currently receiving Eagle plate map data from two LH labs.
Cambridge -
- comes in via sftp - goes into MongoDB & lighthouse_sample table
- SSRs upload manifest with COG UK IDs - samples go into SS, lighthouse_sample table does not get updated with COG UK ID
Milton Keynes -
- comes in via email - does not go into MongoDB or lighthouse_sample table
- SSRs upload manifest with COG UK IDs - samples go into SS

**Comment by KatyTaylor** [comment]
Potential fixes:
- Upload the plate map files that haven't gone through sftp into sftp, to get the data into Mongo / lighthouse_sample
- Manually update the missing cog uk ids in lighthouse_sample
- Implement new feature where specific manifest upload triggers COG UK ID creation and also updates lighthouse_sample

**Comment by KatyTaylor** [comment]
Existing and preferred processes mapped out here - https://lucid.app/lucidchart/invitations/accept/042d2ef6-648c-4278-8462-76eb592dd01a

**Comment by KatyTaylor** [comment]
Proposal:
- Lighthouse labs send us new field: 'cohort'
- This gets stored in Mongo (and lighthouse_sample if wanted)
- This gets set in Sequencescape by Sentinel Sample Creation 

Question:
- Ideally Beckman destination plate endpoint would also set this, for completeness (in case unconsolidated plates arrive and are put through the Beckman process) - would we need Beckman's help to pass another well property through DART, or do we already query Mongo for some metadata about samples?

Assumption:
- It's OK that Sentinel Sample Creation obeys the 'fit to pick' rules. This will be made more flexible by the 'PaM flagging important samples' feature.

**Comment by KatyTaylor** [comment]
Related RT - https://rt.sanger.ac.uk/Ticket/Display.html?id=711817

**Comment by rl15** [comment]
Hold until - for context Rich L wrote (Tuesday, 16 March 2021 at 12:03)   

I believe we can with PAM support (I am optimistic at least). As discussed on the 11th we all want to have all samples cohorts coming via the SFTP. Once we have achieved this we have a static set of eagle samples that were sequenced via sample manifests between 5th Feb and <process improvement date>. Is my understanding this cohort then divides into three sets
 
Samples (Root sample id, plate, well position & result)  we never observed via the SFTP – obvious to patch
Samples (Root sample id, plate, well position & result)  we observed once via SFTP – potentially complicated - are they actually the same sample? – if not then what?
Samples (Root sample id, plate, well position & result)  we observed more than once via SFTP – complicated is rooted in duplicate samples discussions.
 
Correct?
